src/Cake.NuGet/NuGetContentResolver.cs

```diff
@@ -63,15 +63,17 @@ private IReadOnlyCollection<IFile> GetAddinAssemblies(DirectoryPath path, Packag
             var provider = DefaultFrameworkNameProvider.Instance;
             var current = NuGetFramework.Parse(_environment.Runtime.BuiltFramework.FullName, provider);
 
-            // Get all ref assemblies.
+            // Get all ref and runtime assemblies.
             var refAssemblies = _globber.GetFiles(path.FullPath + "/ref/**/*.dll");
+            var runtimeAssemblies = _globber.GetFiles(path.FullPath + "/runtimes/**/*.dll");
```

Comment:
As `IsClrAssembly` looks at the files maybe we should reduce files it should even look at first, something like 
```csharp
GetFiles(path, package, new[] { path.FullPath + "/**/*.dll" })
    .Except(refAssemblies + runtimeAssemblies, pathComparer)
```

And maybe even change `refAssemblies + runtimeAssemblies` to something like
```csharp
var excludedAssemblies = _globber.GetFiles(path.FullPath + "/{ref,runtimes}/**/*.dll");
```

Not tested if it compiles, but gist of it that it could reduce number of directory scans.

Reply:
Sounds like a good idea!